modules/imgproc/test/test_houghlines.cpp

```diff
@@ -262,6 +287,73 @@ void HoughLinesPointSetTest::run_test(void)
     EXPECT_EQ((int)(line_polar_i.at(0).val[2] * 100000.0f), (int)(Theta * 100000.0f));
 }
 
+void StandartHoughLinesTest3f::run_test(bool type)
+{
+    string filename = cvtest::TS::ptr()->get_data_path() + picture_name;
+    Mat src = imread(filename, IMREAD_GRAYSCALE);
+    EXPECT_FALSE(src.empty()) << "Invalid test image: " << filename;
+
+    string xml;
+    if(!type)
+        xml = string(cvtest::TS::ptr()->get_data_path()) + "imgproc/HoughLines2f.xml";
+    else
+        xml = string(cvtest::TS::ptr()->get_data_path()) + "imgproc/HoughLines3f.xml";
+
+    Mat dst;
+    Canny(src, dst, 100, 150, 3);
+    EXPECT_FALSE(dst.empty()) << "Failed Canny edge detector";
+
+    Mat lines;
+    if(!type) {
+        vector<Vec2f> _lines;
+        HoughLines(dst, _lines, rhoStep, thetaStep, threshold, 0, 0);
+        Mat(_lines).copyTo(lines);
+    }
+    else {
+        vector<Vec3f> _lines;
+        HoughLines(dst, _lines, rhoStep, thetaStep, threshold, 0, 0);
+        Mat(_lines).copyTo(lines);
+    }
+
+    String test_case_name = format("lines_%s_%.0f_%.2f_%d_%d_%d", picture_name.c_str(), rhoStep, thetaStep,
+                                   threshold, minLineLength, maxGap);
+    test_case_name = getTestCaseName(test_case_name);
+
+    FileStorage fs(xml, FileStorage::READ);
+    FileNode node = fs[test_case_name];
+    if(node.empty())
+    {
+        fs.release();
+        fs.open(xml, FileStorage::APPEND);
```

Comment:
1. Writing into testdata must be disabled in default builds - put this code under "#ifdef GENERATE_DATA" condition.
Currently this test doesn't check anything, because it rewrites testdata (CI cleanups build folder before every build).
2. Test data should be properly added into opencv_extra repository.

Reply:
1. OK, I added "#ifdef GENERATE_DATA" between 326-331.
2. I'll add test data into opencv_extra repository.

Reply:
I added test data into opencv_extra. Please watch this pull request.
https://github.com/opencv/opencv_extra/pull/460